### DISCUSSION TITLE
Recursive `toResolvedBlocks` for blocks within blocks

### DISCUSSION THREAD

**Discussion by pixelpaulaus** [kirby-tools/community#25] [General]
Is it possible to make toResolvedBlocks recursive? so when there are blocks within blocks it resolves to objects

**Comment by johannschopplich** [comment]
This is possible with a [custom resolver](https://kirby.tools/docs/headless/field-methods#custom-resolvers). You can call `toResolvedBlocks` with in that resolver. Given you have a `prose` block that contains a field `block` of `type: blocks`:

```php
use Kirby\Cms\Block;
use Kirby\Content\Field;

return [
    'blocksResolver' => [
        'resolvers' => [
            'prose:blocks' => fn (Field $field, Block $block) => $field->toResolvedBlocks()
        ]
    ]
];
```
